### ISSUE TITLE
Creating structs in groupby aggregations from the output of `value_counts` raises `exceptions.DuplicateError: multiple fields with name '' found` error

### ISSUE THREAD

**Issue by mmcdermott** [pola-rs/polars#8070]
### Polars version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Issue description

I believe this may be related to https://github.com/pola-rs/polars/issues/7286.

If I have a polars expression that computes the `value_counts` of a column, then builds a new struct from those fields, this works fine in a `select` context but crashes in a `groupby` context with the error `exceptions.DuplicateError: multiple fields with name '' found`. See below:

```
import polars as pl

X = pl.DataFrame({'val': [-3, -2, -1, 0, 1, 2, 3], 'k': [0]*7})

# Works fine
X.groupby('k').agg(
  pl.col('val').value_counts().struct.field('val').alias('val'),
  pl.col('val').value_counts().struct.field('counts').alias('counts')
)

# Raises error `exceptions.DuplicateError: multiple fields with name '' found`
X.groupby('k').agg(
  pl.struct([
    pl.col('val').value_counts().struct.field('val').alias('val'),
    pl.col('val').value_counts().struct.field('counts').alias('counts')
  ])
)
```

For a practical context in which this could be used, I am performing a groupby aggregation which processes not the count of unique values, but the counts of unique deviations from the mean, which requires me to pull apart the `value_counts` struct and piece together a new struct in my code.

### Reproducible example

```python
import polars as pl

X = pl.DataFrame({'val': [-3, -2, -1, 0, 1, 2, 3], 'k': [0]*7})

# Works fine
X.groupby('k').agg(
  pl.col('val').value_counts().struct.field('val').alias('val'),
  pl.col('val').value_counts().struct.field('counts').alias('counts')
)

# Raises error `exceptions.DuplicateError: multiple fields with name '' found`
X.groupby('k').agg(
  pl.struct([
    pl.col('val').value_counts().struct.field('val').alias('val'),
    pl.col('val').value_counts().struct.field('counts').alias('counts')
  ])
)
```

### Expected behavior

I expect the code indicated above to return a struct with two fields, `val`, containing the value and `counts` containing the count, aggregated as a list (though the error persists if aggregations are used atop the expression as well).

### Installed versions

<details>

```
Polars: 0.16.18
Index type: UInt32
Platform: Linux-3.10.0-1160.88.1.el7.x86_64-x86_64-with-glibc2.17
Python: 3.10.4 (main, Mar 31 2022, 08:41:55) [GCC 7.5.0]
```

</details>

**Comment by mmcdermott** [comment]
I also just verified this in a fresh environment with polars 0.17.0 installed via pip. 

Test case:
```
>>> X = pl.DataFrame({'k': [0, 0, 0, 0], 'v': [1, 3, 4, 5]})
>>> val_counts = pl.col('v').value_counts()
>>> new_struct = pl.struct([val_counts.struct.field('v').alias('v'), val_counts.struct.field('counts').alias('counts')])
>>> X.groupby('k').agg(new_struct)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/mmd/Programs/miniconda3/envs/test_polars/lib/python3.10/site-packages/polars/dataframe/groupby.py", line 225, in agg
    self.df.lazy()
  File "/home/mmd/Programs/miniconda3/envs/test_polars/lib/python3.10/site-packages/polars/lazyframe/frame.py", line 1598, in collect
    return wrap_df(ldf.collect())
exceptions.DuplicateError: multiple fields with name '' found
>>> X.select(new_struct)
shape: (4, 1)
┌───────────┐
│ v         │
│ ---       │
│ struct[2] │
╞═══════════╡
│ {4,1}     │
│ {5,1}     │
│ {3,1}     │
│ {1,1}     │
└───────────┘
>>> X.select(new_struct.list())
shape: (1, 1)
┌─────────────────────────┐
│ v                       │
│ ---                     │
│ list[struct[2]]         │
╞═════════════════════════╡
│ [{5,1}, {1,1}, … {3,1}] │
└─────────────────────────┘
>>> X.groupby('k').agg(val_counts)
shape: (1, 2)
┌─────┬─────────────────────────┐
│ k   ┆ v                       │
│ --- ┆ ---                     │
│ i64 ┆ list[struct[2]]         │
╞═════╪═════════════════════════╡
│ 0   ┆ [{5,1}, {3,1}, … {1,1}] │
└─────┴─────────────────────────┘
```

Versions:
```
>>> pl.show_versions()
---Version info---
Polars: 0.17.0
Index type: UInt32
Platform: Linux-5.15.0-69-generic-x86_64-with-glibc2.35
Python: 3.10.10 (main, Mar 21 2023, 18:45:11) [GCC 11.2.0]
---Optional dependencies---
numpy: 1.24.1
pandas: <not installed>
pyarrow: <not installed>
connectorx: <not installed>
deltalake: <not installed>
fsspec: <not installed>
matplotlib: <not installed>
xlsx2csv: <not installed>
xlsxwriter: <not installed>
```

**Comment by mmcdermott** [comment]
This is actually specific to the `value_counts()` output struct, though I don't know why.

**Comment by mmcdermott** [comment]
I just pushed a test for this in my fork that fails, but I don't know how to fix it. I tried changing some things that seemed promising, but I don't know rust so was mostly just guessing, and unsurprisingly, they didn't work. It'd be great if anyone has a chance to take a look at this. My current best theory is that the issue comes about because `value_counts` doesn't just produce a standard struct expression, but rather does some form of internal grouping and re-aggregating internally, and names aren't being appropriately propagated in that step, even though the output expression seems to have output names fully specified at all levels.
